sdks/python/apache_beam/options/pipeline_options_test.py

```diff
@@ -215,6 +216,7 @@ def _add_argparse_args(cls, parser):
       parser.add_argument(
           '--fake_multi_option', action='append', help='fake multi option')
 
+  @pytest.mark.no_xdist
```

Comment:
The setUp and tearDown code was an attempt at fixing the flakiness, by removing side-effects. Perhaps there are additional side-effects we didn't account for?

Reply:
The problem could be related to bad timing when processing tests in parallel, maybe a race condition causing side effects when a test runs before one initial tests call the tearDown method.

Reply:
I think it's very unlikely that parallelization will cause flakiness due to some shared state and a race condition. xdist works by starting up multiple separate python processes each running a partition of the tests. The GIL would eliminate any benefit in a threaded approach.

My guess would be that xdist makes this flaky by sometimes executing some other test that modifies state in the same python process/worker. I'm not sure what exactly no_xdist does, presumably it just causes the test to be run in the main process? It makes sense that that would help since that problematic test is then guaranteed to run in a separate process.

Reply:
no_xdist affects how the test is run by run_pytest.sh. The script was written because we have some tests that don't work well with xdist. Flakiness though isn't a good reason to disable xdist.

I think that this might be the culprit: https://github.com/apache/beam/blob/07bf18a75f3e4b28cd496b83b5aca9bea34f78e1/sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py#L93

Reply:
Actually I'm not sure if it's the culprit.

Reply:
I can't explain it, but I'm pretty sure #14460 is the culprit. Reverting it in #15101 got me three PreCommit runs without a flake in test_display_data.

Reply:
Now that #15115 is merged. Do we still need this change?

Reply:
I think we can close it